temporalio/worker/workflow.py

```diff
@@ -75,6 +71,9 @@ def __init__(
         self._type_hint_eval_str = type_hint_eval_str
         self._running_workflows: Dict[str, WorkflowInstance] = {}
 
+        # Unless there's a truthy TEMPORAL_DEBUG env var, set deadlock timeout
+        self._deadlock_timeout_seconds = None if os.environ.get("TEMPORAL_DEBUG") else 2
```

Comment:
I decided to not use an env var for this in TS and just let users wire in this config however they want.
I know other SDKs have this variable so I guess it's acceptable but I don't really see a good enough reason for this.

Reply:
Yeah, I struggled here because I hate the idea of env var only, but I wanted to be consistent. I think they very intentionally didn't provide an option for disabling the detector in Go when creating the worker.

Reply:
In TS we call it `debugMode` which also changes the workflow runner to run in the main nodejs thread.
I think it's totally fine to expose this as a worker option.

Reply:
Will do

Reply:
Done